engine/deck.go

```diff
@@ -30,3 +30,62 @@ func swap(deck []Card, i int, j int) {
 	deck[j] = deck[i]
 	deck[i] = tmp
```

Comment:
Where is tmp declared?

Reply:
on line 29: `tmp := deck[j]`. The `:=` is shorthand for `var tmp = deck[j]`

Reply:
aw i knew the shorthand i just  didn't see line 29 for some reason